### PR TITLE
feat: [GTM-579]: Self Service Credit Cards with Stripe

### PR DESCRIPTION
**Pull request by thomas-grasman**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/34012)

### CODE REVIEW

**Comment by thomas-grasman** [comment]
trigger messagecheck

**Comment by thomas-grasman** [comment]
trigger messagecheck

File: 120-ng-manager/src/main/java/io/harness/serializer/NextGenRegistrars.java [hunk 1]
```diff
@@ -36,96 +36,97 @@
...
-                   .build())
-          .add(YamlSchemaRootClass.builder()
-                   .entityType(EntityType.INFRASTRUCTURE)
-                   .availableAtProjectLevel(true)
-                   .availableAtAccountLevel(true)
-                   .availableAtOrgLevel(true)
-                   .clazz(InfrastructureConfig.class)
-                   .build())
-          .build();
+          ImmutableList.<YamlSchemaRootClass>builder()
```
Comment (pranil-harness) [inline 120-ng-manager/src/main/java/io/harness/serializer/NextGenRegistrars.java:82]: @thomas-grasman - do we need these changes?

**Review by pranil-harness — COMMENTED** [review]
> Reviewed 7 of 67 files at r4.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34012)* status: 60 file(s) to review. 0 of 1 approvals obtained.

**Review by pranil-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r1, 8 of 59 files at r3, 1 of 67 files at r7, 1 of 66 files at r10, 58 of 67 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34012)* status: :shipit: complete! ✓

**Pull request merged**